README.md

```diff
@@ -151,7 +151,9 @@ By default, the Kogito services are built with traditional `java` compilers to s
 
 Kogito services implemented with [Quarkus](https://quarkus.io/guides/kogito-guide) can be built to native binary. This means very low footprint on runtime (see [performance examples](https://www.graalvm.org/docs/examples/java-performance-examples/)), but a lot of resources during build time. For more information about AOT compilation, see [GraalVM Native Image](https://www.graalvm.org/docs/reference-manual/aot-compilation/).
 
-In Kogito Operator tests, native builds take approximately 10 minutes and the build pod can consume up to 10GB of RAM and 1.5 CPU cores. Ensure that you have these resources available when running native builds.
+In Kogito Operator tests, native builds take approximately 10 minutes and the build pod can consume up to 10GB of RAM and 1.5 CPU cores.
+
+:warning: Ensure that you have these resources available on your OpenShift nodes when running native builds. Otherwise the S2I build will fail.
```

Comment:
Do you mind to do some research on this topic? At the time I wrote this, we didn't have too much info, but things could change. Maybe a GraalVM documentation or Stack Overflow discussions regarding Native compilation resource consumption?

Reply:
Sure, will take a look.

Reply:
Could you also instruct user how to set it?

Reply:
About the Native Image resource consumption - I was reading a [GraalVM documentation](https://www.graalvm.org/docs/reference-manual/native-image/#prerequisites) where they list:

> Another prerequisite to consider is the maximum heap size. Physical memory for running a JVM-based application may be insufficient to build a native image. For server-based image building we allow to use 80% of the reported physical RAM for all servers together, but never more than 14GB per server (for exact details please consult the native-image source code). If you run with --no-server option, you will get the whole 80% of what is reported as physical RAM as the baseline. This mode respects -Xmx arguments additionally.

Not should what the server-based image is, the documentation is very limited in that case, but it should start a background daemon with some caching capabilities. What I know is that according to the documentation Quarkus has the property `quarkus.native.enable-server` set to false by default. So by that we are using up to 80% of all memory available, i.e. 14GB limit is not enforced.

However, when it comes to CPU utilization, there seems to be no way on limiting that and according to some posts on Stack Overflow it consumes as much cores as possible.

Reply:
I think this worth more in depth research/tests. @sutaakar could you start a native build with the `KOGITO_OPTS` env var adding this property `-Dquarkus.native.enable-server=true` and observe its behavior? wdyt?

Reply:
@ricardozanini 
I tried to build jbpm-quarkus-example on branch 0.7.0 with KOGITO_OPTS set to '-Dquarkus.native.enable-server=true'.
The build repeatedly crashed on storing container layers:
```
STEP 10: COMMIT temp.builder.openshift.io/ksuta-test/example-quarkus-builder-6:4d81515b
Getting image source signatures
Copying blob sha256:d8e1f35641acb80b562f70cf49911341dfbe8c86f4d522b18efbf3732aa74223
Copying blob sha256:b6f081e4b2b6de8be4b1dec132043d14c121e968384dd624fb69c2c07b482edb
Copying blob sha256:359e1f42a53f7d20d1cf84792849442ab5a37da0508b3397e7ffb14062144378
error: build error: error committing container for step {Env:[LIMIT_MEMORY= LIMIT_CPU= NATIVE=true KOGITO_OPTS=-Dquarkus.native.enable-server=true OPENSHIFT_BUILD_COMMIT=3e1b948b430219895253f3718f7c6e4168b2c4bb OPENSHIFT_BUILD_REFERENCE=0.7.0 OPENSHIFT_BUILD_SOURCE=https://github.com/kiegroup/kogito-examples OPENSHIFT_BUILD_NAMESPACE=ksuta-test OPENSHIFT_BUILD_NAME=example-quarkus-builder-6 PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin container=oci KOGITO_HOME=/home/kogito USER=kogito GRAALVM_HOME=/usr/share/graalvm GRAALVM_VERSION=19.2.1 JAVA_HOME=/usr/lib/jvm/java-1.8.0 JAVA_VENDOR=openjdk JAVA_VERSION=1.8.0 MAVEN_HOME=/usr/share/maven MAVEN_VERSION=3.6.0 S2I_MODULE_LOCATION=/usr/local/s2i KOGITO_VERSION=0.7.0          OPENSHIFT_BUILD_NAME=example-quarkus-builder-6 OPENSHIFT_BUILD_NAMESPACE=ksuta-test OPENSHIFT_BUILD_SOURCE=https://github.com/kiegroup/kogito-examples OPENSHIFT_BUILD_REFERENCE=0.7.0 OPENSHIFT_BUILD_COMMIT=3e1b948b430219895253f3718f7c6e4168b2c4bb KOGITO_OPTS=-Dquarkus.native.enable-server=true NATIVE=true LIMIT_CPU= LIMIT_MEMORY=] Command:cmd Args:[/usr/local/s2i/run] Flags:[] Attrs:map[] Message:CMD /usr/local/s2i/run Original:CMD /usr/local/s2i/run}: error copying layers and metadata for container "e62c1f2ba8b610a7b2098348ba86b09d93a9b078a49bc0fdec0ec1a1f5e1f9c2": Error reading blob sha256:99c07e395e425baf86c84598d0b1154f94736f33725995a23a7c0439bb0c8841: error reading blob from source image "containers-storage:kogito-quarkus-ubi8-s2i-working-container": error opening file "/tmp/buildah085350749/sha256:99c07e395e425baf86c84598d0b1154f94736f33725995a23a7c0439bb0c8841" to buffer layer blob: open /tmp/buildah085350749/sha256:99c07e395e425baf86c84598d0b1154f94736f33725995a23a7c0439bb0c8841: no such file or directory
```

When running without the property the build is successful.

Reply:
Could you open a JIRA to investigate it? Thanks!

Reply:
Created https://issues.redhat.com/browse/KOGITO-1397